python/ray/experimental/channel/nccl_group_interface.py

```diff
@@ -0,0 +1,76 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Callable, Optional, Tuple
+
+import ray
+from ray.util.annotations import DeveloperAPI
+
+if TYPE_CHECKING:
+    import torch
+
+
+# Signature for a torch.Tensor allocator is:
+# (shape: Tuple[int], dtype: torch.dtype) -> torch.Tensor.
+TorchTensorAllocator = Callable[[Tuple[int], "torch.dtype"], "torch.Tensor"]
+
+
+@DeveloperAPI
+class NcclGroupInterface(ABC):
```

Comment:
does it have to be nccl? I.e., isn't it going to work with whatever interface that supports send/recv?

Reply:
Good point. We can make it general.